### ISSUE TITLE
Allow multiple banned-import-groups within a single enforcer rule

### ISSUE THREAD

**Issue by skuzzle** [skuzzle/restrict-imports-enforcer-rule#9]
This would allow influences of rules to other rules. My favorite: 
* A source file is only subject to a single group
* The group to which a source file is subject is chosen by the longest `basePackage` pattern which matches the source file

```xml
...
<bannedImportGroups>
    <bannedImportGroup>
        <basePackage>com.foo.**</basePackage>
        <allowedImports>...</allowedImports>
    </bannedImportGroup>
    <bannedImportGroup>
        <basePackage>com.foo.bar.**</basePackage>
        <allowedImports>...</allowedImports>
    </bannedImportGroup>
</bannedImportGroups>
...
```

In this sample, source files from `com.foo.bar` would only be subject to the rules within the second group.

Another approach could be that all groups are applied to a single source file and groups with a more specific `basePackage` can override rules from groups with less specific `basePackage`.

**Comment by skuzzle** [comment]
As we have multiple base package, its required to define the relation which states wheter one group is more specific than another.

Rules:
1. Specifity is only relevant if both compared groups _share_ as base package
1.1. Groups `g1` and `g2` share as base package if either:
    * Any base package of `g1` matches at least one base package of `g2`
    * Any base package of `g1` is a prefix of at least one base package of `g2`

2. A pattern without wildcards is always more specific than a package including wild cards

**Comment by skuzzle** [comment]
Released as 0.13.0
